content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -275,33 +275,79 @@ node {
 }
 ----
 
-=== Defining steps
 
+=== Defining global variables
+
+Internally, scripts in the `vars` directory are instantiated as a singleton
+on-demand, when used first.  So it is possible to define more methods,
+properties on a single file that interact with each other:
+
+[source,groovy]
+----
+// vars/acme.groovy
+def setFoo(v) {
+    this.foo = v;
```

Comment:
There's a weird behavior in Pipeline scripts when variables/fields are the same name as the getter (resulting in stack overflows) – does this not apply when using `this`, or when in a global library?